bindings/node/lib/clientEncryption.js

```diff
@@ -549,6 +554,75 @@ module.exports = function (modules) {
       return value;
     }
 
+    /**
+     * Creates a collection that has encrypted document fields
+     *
+     * @template {TSchema} - Schema for the collection being created
+     * @param {Db} db - A database to create the collection in
+     * @param {string} name - The name of the collection to be created
+     * @param {object} options - Options for createDataKey and for createCollection.
+     * @param {string} options.provider - provider name
+     * @param {ClientEncryptionCreateDataKeyProviderOptions} options.createDataKeyOptions - options to pass to createDataKey
+     * @param {CreateCollectionOptions} options.createCollectionOptions - options to pass to createCollection, must include `encryptedFields`
+     * @returns {Promise<{ collection: Collection<TSchema>, encryptedFields: Document }>} - created collection and generated encryptedFields
+     * @throws {MongoCryptCreateEncryptedCollectionError} - If part way through the process createDataKey fails, an error will be rejected that has the `encryptedFields` that were created.
+     */
+    async createEncryptedCollection(db, name, options) {
+      const {
+        provider,
+        createDataKeyOptions = {},
+        createCollectionOptions: {
+          encryptedFields: { ...encryptedFields },
+          ...createCollectionOptions
+        }
+      } = options;
+
+      if (Array.isArray(encryptedFields.fields)) {
+        encryptedFields.fields = [...encryptedFields.fields];
+
+        const createDataKeyPromises = [];
+        for (const [index, field] of encryptedFields.fields.entries()) {
+          if (typeof field !== 'object' || field.keyId != null) {
+            continue;
+          }
+
+          createDataKeyPromises.push(
+            (async () => {
+              const keyId = await this.createDataKey(provider, createDataKeyOptions);
+              return { nullKeyIdOffset: index, keyId };
```

Comment:
Fwiw, I think this might be a bit more readable if this was also just `index` rather than `nullKeyIdOffset` :)